recipe-server/normandy/settings.py

```diff
@@ -110,6 +111,25 @@ class Core(Configuration):
         }
     }
 
+    # Content Security Policy
+    def CSP_DEFAULT_SRC(self):
+        srcs = ["'self'"]
+        if self.CDN_URL:
+            srcs.append(self.CDN_URL)
+        return srcs
+    CSP_OBJECT_SRC = "'none'"  # not using <object>, <embed>, and <applet> elements
+    CSP_WORKER_SRC = "'none'"  # not using JS Worker, SharedWorker, or ServiceWorkers
+    CSP_FRAME_SRC = "'none'"  # not using frames or iframes
+
+    # these don't fallback to default-src
+    CSP_BASE_URI = "'none'"  # not using <base>
+    CSP_FRAME_ANCESTORS = "'none'"  # this page isn't iframed elsewhere
+    CSP_FORM_ACTION = "'self'"  # we only submit forms to ourselves
+
+    # when https is enabled
+    CSP_UPGRADE_INSECURE_REQUESTS = True
+    CSP_BLOCK_ALL_MIXED_CONTENT = True
```

Comment:
I bet you need to be a little more careful with this: not all of our configs use https. In particular, you'll need to change `CSP_UPGRADE_INSECURE_REQUESTS` in `ProductionInsecure`, at least. I think the compose set up uses `ProductionInsecure`.